compiler/rustc_lint/src/context.rs

```diff
@@ -50,6 +50,10 @@ use std::cell::Cell;
 use std::iter;
 use std::slice;
 
+type EarlyLintPassFactory = dyn Fn() -> EarlyLintPassObject + sync::Send + sync::Sync;
+type LateLintPassFactory =
+    dyn for<'tcx> Fn(TyCtxt<'tcx>) -> LateLintPassObject<'tcx> + sync::Send + sync::Sync;
```

Comment:
Is the input parameter necessary? It might be nice to maintain that lint passes are constructed without access to anything.

Reply:
Yes. otherwise you get:
```None
error[E0582]: binding for associated type `Output` references lifetime `'tcx`, which does not appear in the trait input types
```